src/lib/render.ts

```diff
@@ -50,6 +50,73 @@ export const render =
                                ...options,
                              }));
         part.appendInto(container);
+        console.log('render container', (container as HTMLElement).innerHTML);
+      }
+      part.setValue(result);
+      part.commit();
+    };
+
+export const hydrate =
+    (result: unknown,
+     container: Element|DocumentFragment,
+     options?: Partial<RenderOptions>) => {
+      console.log('hydrate', container.childNodes[0].textContent);
+
+      let part = parts.get(container);
+
+      if (part === undefined) {
+        let rootPart: PartInfo|undefined = undefined;
+        const partStack: PartInfo[] = [];
+        const walker = document.createTreeWalker(
+            container, NodeFilter.SHOW_COMMENT, null, false);
+        let node: Comment|null;
+        while ((node = walker.nextNode() as Comment | null) !== null) {
+          if (node.nodeType === Node.COMMENT_NODE) {
+            if (node.textContent!.startsWith('lit-part')) {
+              // This is an NodePart opening marker
+              const partInfo = {
+                startNode: node,
+                endNode: undefined as unknown as Node,
+              };
+              if (partStack.length > 0) {
+                const parentInfo = partStack[partStack.length - 1];
+                if (parentInfo.children === undefined) {
+                  parentInfo.children = [];
+                }
+                parentInfo.children.push(partInfo);
+              } else {
+                console.assert(
+                    rootPart === undefined,
+                    'there should be exactly one root part in a render container');
+                rootPart = partInfo;
```

Comment:
This'll mean you can't render something like a `repeat` inside the container. Why not just initialize a root part to begin with?